tools/tertiary-analysis/scmap/scmap_index_cluster.xml

```diff
@@ -6,9 +6,13 @@
     <expand macro="requirements" />
     <command detect_errors="exit_code"><![CDATA[
         scmap-preprocess-sce.R --input-object "${input_single_cell_experiment}" --output-sce-object "${input_single_cell_experiment}.preprocessed" && scmap-index-cluster.R --input-object-file "${input_single_cell_experiment}" --cluster-col '$cluster_col' --output-object-file '$output_single_cell_experiment' --output-plot-file '$plot'
+        #if $train_id
+        --train-id '${train_id}' 
+        #end if
     ]]></command>
     <inputs>
         <param type="data" name="input_single_cell_experiment" label="SingleCellExperiment object" format="rdata" help="File with serialized SingleCellExperiment object as produced by 'scmap select features'" />
+        <param type="text" name="train_id" label="dataset ID" help="ID of the training dataset" />
```

Comment:
Isn't there anyway of smuggling these dataset set identifiers inside the SCE object and then grabbing them from there?

Reply:
This is exactly what is done here :)

Reply:
So then other usages of Dataset ID around the other tools are just to fish it outside of the objects?

Reply:
@pcm32 Yes, because the plan is to have a library of pre-trained classifiers that will provide predictions for new dataset. It would be really convenient if these classifiers stored information on what data they were trained on.